go-controller/pkg/ovn/gateway_init.go

```diff
@@ -126,7 +126,7 @@ func (oc *DefaultNetworkController) gatewayInit(nodeName string, clusterIPSubnet
 	}
 
 	if oc.loadBalancerGroupUUID != "" {
```

Comment:
same thing here, we seem to be coupling `routerLoadBalancerGroupUUID` to be set if `loadBalancerGroupUUID` is set which is probably true - but maybe if `gatewayMode` is not enabled then we don't have any routers to begin with and we don't need this I guess, but at least a comment or an explicit check will ensure devs don't trip over this unexplained dependency if some refactor happens in the future

Reply:
Ok.